crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -402,12 +597,20 @@ impl<F, T> TryFrom<ResponseRouterData<F, NovalnetPaymentsResponse, T, PaymentsRe
                     .transaction
                     .clone()
                     .and_then(|data| data.tid.map(|tid| tid.expose().to_string()));
+
+                let mandate_reference_id = NovalnetPaymentsResponseTransactionData::get_token(
+                    item.response.transaction.clone().as_ref(),
+                );
+
                 let transaction_status = item
                     .response
                     .transaction
+                    .as_ref()
                     .and_then(|transaction_data| transaction_data.status)
                     .unwrap_or(if redirection_data.is_some() {
                         NovalnetTransactionStatus::Progress
+                    } else if mandate_reference_id.is_some() {
+                        NovalnetTransactionStatus::OnHold
```

Comment:
Can you please explain the login here ?

Reply:
this code is redundant, I have removed it